scripts/install_prereqs.py

```diff
@@ -0,0 +1,453 @@
+    """
+    system_packages_to_install = []
+    # Install ninja and/or ccache if needed
+    # check if ninja is already installed before attempting to install it
+    if args.generator == 'Ninja':
+        if is_tool_installed('ninja'):
+            print('Ninja requested but is already installed. Skipping installation...')
+        elif is_linux_os():
+            system_packages_to_install.append('ninja-build')
+        elif is_mac_os:
+            system_packages_to_install.append('ninja')   
+
+    # ccache is supported only on mac and linux and check if it is already installed
+    if args.ccache and (is_linux_os() | is_mac_os()) and not is_tool_installed('ccache'):
+        if is_tool_installed('ccache'):
+            print('ccache requested but is already installed. Skipping installation...')
+        elif is_linux_os() or is_mac_os():
+            system_packages_to_install.append('ccache')
+
+    if system_packages_to_install:
+        install_system_packages(system_packages_to_install)
+
+
+def step_install_python_packages(args):
+    """ Install python packages (dependencies)
+
+    Args:
+        args (`argparse.Namespace`): Parsed command line arguments
+    """
+    # Check for packages installed in this python environment
+    installed_packages = {pkg.key for pkg in pkg_resources.working_set}
+
+    # Get the list of packages we intend to install
+    requirements_file_path = get_pip_requirements_file_path()
+    packages_to_install = None
+    with open(requirements_file_path, 'r') as requirements_file:
+        packages_to_install = {line.strip('\n') for line in requirements_file.readlines()}
+
+    # Check if there is any work to do by comparing these lists
+    if packages_to_install and not (packages_to_install-installed_packages):
+        print('Skipping python packages installation because following '\
+               'python packages are already installed: \n {0}'.format(' '.join(packages_to_install)))
+        return
+
+    install_python_packages()
+
+
+def step_install_cpp_packages(args):
+    """ Install C++ packages (dependencies)
+
+    Installation supported only via vcpkg as system package manager can affect
+    system wide packages. User should handle those themselves
+
+    Args:
+        args (`argparse.Namespace`): Parsed command line arguments
+    """
+    if not args.no_vcpkg:
+        # Check if vcpkg is already installed
+        found_vcpkg_executable = os.path.exists(get_vcpkg_executable_file_path())
+
+        # Install packages and only install vcpkg if its not built already
+        install_packages_with_vcpkg(args.arch, build_vcpkg=not found_vcpkg_executable)
+
+    else:
+        # Do not install dependencies automatically as we want users to have control
+        # on their system level package installations
+        print('Please install packages listed in {0}'\
+              'with your system package manager'.format(get_vcpkg_response_file_path(get_vcpkg_triplet(args.arch))))
```

Comment:
I agree things would be simple if we decide to just use vcpkg.
Most of the code is an optimization. It will shave off some time (few minutes) on Github runners and super helpful for local dev workflows.